app/assets/src/components/views/discovery/DiscoverySidebar.jsx

```diff
@@ -203,29 +202,33 @@ export default class DiscoverySidebar extends React.Component {
     });
   }
 
-  renderMetadataRowBlock(rows, total) {
+  renderMetadataRowBlock(rows, total, field) {
+    const { onFilterClick } = this.props;
     return rows.map((entry, i) => {
       const { count, text, value } = entry;
       const percent = Math.round(100 * count / total, 0);
+      const onClick = () => {
+        onFilterClick && onFilterClick(field, value);
+        logAnalyticsEvent("DiscoverySidebar_metadata-filter_clicked", {
+          value,
+          count,
+          percent,
+          rows: rows.length,
+        });
+      };
       return (
         <div className={cs.barChartRow} key={`${value}_row_${i}`}>
           <dt className={cs.barLabel} key={`${value}_label_${i}`}>
-            <a
-              onClick={() => {
-                this.handleFilterClick(value);
-                logAnalyticsEvent("DiscoverySidebar_metadata-filter_clicked", {
-                  value,
-                  count,
-                  percent,
-                  rows: rows.length,
-                });
-              }}
-            >
+            <a onClick={onClick}>
```

Comment:
I think `withAnalytics` is the actual function defined for this.
Funny thing is that withAnalytics also comes with a lot more "event name" validation... 

I will leave up to you to update the code to use that function.

Reply:
ah yes, it looks like withAnalytics is preferred now so I switched to it 👍